src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,339 @@
+import os
+import numpy as np
+from time import sleep
+
+try:
+    import clr
+
+    lf_root = os.environ['LIGHTFIELD_ROOT']
+    automation_path = lf_root + '\PrincetonInstruments.LightField.AutomationV4.dll'
+    addin_path = lf_root + '\AddInViews\PrincetonInstruments.LightFieldViewV4.dll'
+    support_path = lf_root + '\PrincetonInstruments.LightFieldAddInSupportServices.dll'
+
+    addin_class = clr.AddReference(addin_path)
+    automation_class = clr.AddReference(automation_path)
+    support_class = clr.AddReference(support_path)
+
+    import PrincetonInstruments.LightField as lf
+
+    clr.AddReference("System.Collections")
+    clr.AddReference("System.IO")
+    from System.Collections.Generic import List
+    from System import String
+    from System.IO import FileAccess
+except Exception as e:
+    print(f"Exception occurred during import: {e}")
+
+
+class LightfieldApp:
+    def __init__(self, visible):
+
+        self._addinbase = lf.AddIns.AddInBase()
+        self._automation = lf.Automation.Automation(visible, List[String]())
+        self._application = self._automation.LightFieldApplication
+        self._experiment = self._application.Experiment
+
+    @property
+    def automation(self):
+        return self._automation
+
+    @property
+    def addinbase(self):
+        return self._addinbase
+
+    @property
+    def application(self):
+        return self._application
+
+    @property
+    def experiment(self):
+        return self._experiment
+
+    def __del__(self):
+        """
+        Uses Python garbage collection method used to terminate
+        AddInProcess.exe, if this is not done, LightField will not reopen
+        properly.
+        """
+        self.automation.Dispose()
+        print('Closed AddInProcess.exe')
+
+    def set(self, setting, value):
+        """
+        Helper function for setting experiment parameters with basic value
+        checking.
+        """
+
+        if self.experiment.Exists(setting):
+            if self.experiment.IsValid(setting, value):
+
+                self.experiment.SetValue(setting, value)
+
+            else:
+
+                # TODO: might need to add a proper exception
+                print('Invalid value: {} for setting: {}.'.format(value, setting))
+
+        else:
+
+            # TODO: might need to add a proper exception
+            print('Invalid setting:{}'.format(setting))
+
+        return
+
+    def get(self, setting):
+        """
+        Helper function for getting experiment parameters with a basic check
+        to see if a specific setting is valid.
+        """
+
+        if self.experiment.Exists(setting):
+
+            value = self.experiment.GetValue(setting)
+
+        else:
+
+            value = []
+            print('Invalid setting: {}'.format(setting))
+
+        return value
```

Comment:
remove lines

```
if self.experiment.Exists(setting):
    value = self.experiment.GetValue(setting)
else:
    value = []
    print('Invalid setting: {}'.format(setting))
return value

Reply:
I am a bit confused, this is what helps me pull the certain values that I need for the experiment and store it in the list. The program will not function without this.